src/libraries/Microsoft.PowerFx.Interpreter/Functions/Library.cs

```diff
@@ -545,6 +565,26 @@ internal static partial class Library
                     returnBehavior: ReturnBehavior.ReturnBlankIfAnyArgIsBlank,
                     targetFunction: ForAll)
             },
+            {
+                BuiltinFunctionsCore.GUIDPure,
```

Comment:
There's a tricky bit here: Guids are implicitly coercible to strings, but we don't mark that in the IR. We tried doing this earlier, but had to back it out due to compat concerns with Canvas: #240 / #245

Could be fine right now to be limited, but note that we'll have pretty big gaps, test cases like:
>> "TheGuidIs: " & GUID("SOME-VALID-GUID")

pass in JS, not C#